netbout-web/src/main/java/com/netbout/email/EmCatch.java

```diff
@@ -0,0 +1,170 @@
+ * this code accidentally and without intent to use it, please report this
+ * incident to the author by email.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+package com.netbout.email;
+
+import com.jcabi.aspects.Immutable;
+import com.jcabi.aspects.Loggable;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.NoSuchProviderException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.FlagTerm;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+/**
+ * Email Catch.
+ * @author Erim Erturk (erimerturk@gmail.com)
+ * @version $Id$
+ * @since 2.15
+ */
+@Immutable
+@Loggable(Loggable.DEBUG)
+@ToString(of = { "action", "user", "password" })
+@EqualsAndHashCode(of = { "action", "user", "password" })
+final class EmCatch {
+    /**
+     * Action.
+     */
+    private final transient Action action;
+    /**
+     * Email Login key.
+     */
+    private final transient String user;
+    /**
+     * Email password.
+     */
+    private final transient String password;
+    /**
+     * Email server host.
+     */
+    private final transient String host;
+    /**
+     * Email server port.
+     */
+    private final transient int port;
+    /**
+     * Ctor.
+     * @param act Email message handler
+     * @param usr Email login key
+     * @param pass Email password
+     * @param hst Email server host
+     * @param prt Email server port
+     * @param period Email server check period
+     * @checkstyle ParameterNumberCheck (3 lines)
+     */
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    EmCatch(final Action act, final String usr, final String pass,
+            final String hst, final int prt, final long period) {
+        this.action = act;
+        this.user = usr;
+        this.password = pass;
+        this.host = hst;
+        this.port = prt;
+        final Thread monitor = new Thread(
+            new Runnable() {
+                @Override
+                public void run() {
+                    while (true) {
+                        EmCatch.this.check();
```

Comment:
@erimerturk @pinaf If an exception occurs here we should just print it to the log.
in my opinion no reason to interrupt this thread.